### DISCUSSION TITLE
Allow Custom Order of Bookmarks

### DISCUSSION THREAD

**Discussion by ched-dev** [directus/directus#21141] [Feature Requests]
### Summary

I'd like to be able to set a custom order of my bookmarks within a content collection. Right now it is sorted alphabetically.

### Motivation

As my bookmarks list gets longer, I'd like to keep my more frequently used bookmarks near the top.

### Detailed Design

A basic custom sort order will do. No need to overthink it :)

### Must Have:
- Ability to change the order priority of a bookmark
- Sort order is the same for all users

### Could Have:
- An inline changer, or a modal to move them around (your choice)

### Won't Have:
- Personal (user-specific) sorting

### Drawbacks

Seems like a low-risk high-reward feature. Can be considered a lower priority.

### Alternatives

I currently can add numbers to the beginning as a work around, but I just don't like doing that.

### Adoption Strategy

As a new feature it won't break existing installations. Users can opt-in to using it at any time.

**Comment by joppehoekstra** [comment]
Would love to see this feature!

**Comment by sumpuck** [comment]
Workaround: Custom Bookmark Ordering Using Invisible Unicode Characters
Hey everyone! 👋
While waiting for the official drag & drop sorting feature for bookmarks, I found a working workaround that allows custom ordering without cluttering the UI with numbers or letters.
Solution: Zero Width Space Characters (U+200B)
The idea is to prefix bookmark names with invisible Unicode characters. Since bookmarks are sorted alphabetically, we can use different amounts of Zero Width Space characters to control the order.
How it works:

1st bookmark: ​Bookmark Name (1 × U+200B)
2nd bookmark: ​​Bookmark Name (2 × U+200B)
3rd bookmark: ​​​Bookmark Name (3 × U+200B)
And so on...

Ready-to-use template for 1-10 bookmarks:
​Active Orders
​​Drafts
​​​Under Review
​​​​In Production
​​​​​Ready to Ship
​​​​​​Shipped
​​​​​​​Delivered
​​​​​​​​Returns
​​​​​​​​​Archive
​​​​​​​​​​Deleted
For copy-paste (replace with your bookmark names):
​[Your Bookmark 1]
​​[Your Bookmark 2]
​​​[Your Bookmark 3]
​​​​[Your Bookmark 4]
​​​​​[Your Bookmark 5]
​​​​​​[Your Bookmark 6]
​​​​​​​[Your Bookmark 7]
​​​​​​​​[Your Bookmark 8]
​​​​​​​​​[Your Bookmark 9]
​​​​​​​​​​[Your Bookmark 10]
Benefits:

✅ Completely invisible - no visual clutter
✅ Works reliably - tested and confirmed working in Directus 11.8.0
✅ No database modifications needed
✅ Easy to implement - just copy & paste as plain text
✅ Maintains alphabetical sorting while achieving custom order

✅ Confirmed Working:
This method has been tested and verified to work in Directus 11.8.0 when pasted as unformatted text.
Technical Details:

Character: Zero Width Space (U+200B)
HTML Entity: &#8203;
Hex Code: \u200B
How to type: Alt + 8203 (Windows)

Usage Instructions:

Copy the appropriate line from the template above
Replace [Your Bookmark X] with your actual bookmark name
Important: Paste as plain text (Ctrl+Shift+V or "Paste as plain text") into the "Name" field when creating bookmarks in Directus
Repeat for all bookmarks

⚠️ Critical Note: The invisible characters only work when pasted as unformatted/plain text. Regular paste (Ctrl+V) may strip the invisible characters.
How to paste as plain text:

Windows/Linux: Ctrl+Shift+V
Mac: Cmd+Shift+V
Alternative: Right-click → "Paste as plain text" or "Paste without formatting"

Result:
Your bookmarks will appear in the sidebar in the exact order you want (1→10), even though Directus sorts them alphabetically. The invisible characters are completely unnoticeable to end users.

Note: This is a temporary workaround while we wait for the official sorting feature. It's particularly useful for teams who need custom bookmark ordering right now.
Hope this helps the community! Let me know if you have any questions or improvements to this approach.
References:

Related to issues: #22576, #21141, #23077
Unicode Standard: [Zero Width Space](https://unicode.org/charts/PDF/U2000.pdf)

**Comment by joggienl** [comment]
Interesting request!
I think it would be very neat to be able to order bookmarks per collection.

The tip from @sumpuck looks valid, I suppose you could test that if you want. The issue is that this approach is pretty much invisible hacking: it breaks easily and doesn't help if you're using translation strings for your bookmarks.

However, I found an actual visible hack that works (for me): use numbers in the translation keys. The bookmarks get ordered based on the key values, not what they actually translate to.

For example, consider these two bookmarks with translated titles:

1. `$t:blogs` translated to `ZZ Blogs`
2. `$t:news` translated to `AA News`

You might expect `AA News` to appear above `ZZ Blogs` due to alphabetical sorting, but it doesn't because the sorting happens on the translation key, not the translated value!

This is actually useful because we can embed ordering numbers into the keys:

1. `$t:9_blogs` translated to `Blogs`
2. `$t:1_news` translated to `News`

Now `News` will render before `Blogs`, and you won't see any numbers in the actual display names.
